### ISSUE TITLE
Add documentation for creating and using new templates

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#3292]
### Context

For users who would like to use other templates than the PAM50 template, there should be clear documentation on:
- how to generate those templates;
- how to use them.

Notably
- the template files should be placed under a subfolder template/
- the file `info_label.txt` should be present
  - this file should follow conventions that are used by `sct_register_to_template`. See notably: 

    https://github.com/neuropoly/spinalcordtoolbox/blob/60ea08c9d664cf403f2982f8f5c8ae1cc5d704a3/spinalcordtoolbox/scripts/sct_register_to_template.py#L335-L342

Related to: #2423, https://github.com/sct-data/exvivo-template/issues/5,

**Comment by joshuacwnewton** [comment]
If I recall correctly, recently there are/were students in the lab focusing on creating templates/atlases. (And so, they have had to "rediscover" the process of generating templates via older template-generation code and repos, I think created by @benjamindeleener?)

If we're currently revisiting the process of generating templates, perhaps it would be good to keep this issue in mind, and make sure to document the process as we go along?

**Comment by jcohenadad** [comment]
Absolutely! Here is the relevant repository: https://github.com/neuropoly/template

Actively updated by @NadiaBlostein and @rohanbanerjee
